operator/src/lib.rs

```diff
@@ -91,6 +95,50 @@ fn count_pods(hashed: &HashMap<String, Vec<Pod>>) -> usize {
 }
 
 impl SparkState {
+    /// Process the cluster status:
+    /// If no status is set, write image, version and timestamp to status
+    /// Check for restart, start, stop commands
+    pub async fn process_cluster_status(&mut self) -> SparkReconcileResult {
+        if let Some(status) = &self.status {
+            if status.version != self.spec.version {
+                info!(
+                    "Current version [{}] - required version [{}] -> updating cluster...",
+                    status.version, &self.spec.version
+                );
+
+                for pod in &self.context.list_pods().await? {
```

Comment:
I'm fine with leaving it like this just to note that this is different from how I do it (not saying that my way is better).
I set target_version and current_version and then in my reconcile_pods method check for each pod if they differ and delete the pod and then requeue.